desc/plotting.py

```diff
@@ -1649,42 +1848,84 @@ def plot_comparison(
         equal=True,
     )
     ax = np.atleast_1d(ax).flatten()
+
+    plot_data = {}
+    for string in [
+        "rho_R_coords",
+        "rho_Z_coords",
+        "vartheta_R_coords",
+        "vartheta_Z_coords",
+    ]:
+        plot_data[string] = []
     for i, eq in enumerate(eqs):
-        fig, ax = plot_surfaces(
-            eq,
-            rho,
-            theta,
-            zeta,
-            ax,
-            theta_color=colors[i % len(colors)],
-            theta_ls=linestyles[i % len(linestyles)],
-            theta_lw=lws[i % len(lws)],
-            rho_color=colors[i % len(colors)],
-            rho_ls=linestyles[i % len(linestyles)],
-            rho_lw=lws[i % len(lws)],
-            lcfs_color=colors[i % len(colors)],
-            lcfs_ls=linestyles[i % len(linestyles)],
-            lcfs_lw=lws[i % len(lws)],
-            axis_color=colors[i % len(colors)],
-            axis_alpha=0,
-            axis_marker="o",
-            axis_size=0,
-            label=labels[i % len(labels)],
-            title_font_size=title_font_size,
-            xlabel_fontsize=xlabel_fontsize,
-            ylabel_fontsize=ylabel_fontsize,
-        )
+        if return_data:
```

Comment:
might be cleaner to always return data from plot_surfaces here and only return it from plot_comparison if asked (just to avoid big duplicated function call), but also ok how it is.